### PR TITLE
Some more code cleanup

### PR DESCRIPTION
**Pull request by wborn**
This cleanup includes:

* Use enhanced for loops
* Use text blocks
* Use Objects.equals
* Fix some typos
* Remove redundant variable initialization
* Remove redundant null checks with instanceof
* Remove redundant thrown Exceptions
* Remove redundant empty String concatenation

### CODE REVIEW

File: bundles/org.openhab.core.ui.icon/src/test/java/org/openhab/core/ui/icon/internal/IconServletTest.java [hunk 1]
```diff
@@ -66,7 +66,7 @@ public boolean isReady() {
         }
 
         public String getOutput() {
-            return new String(outputStream.toByteArray());
+            return outputStream.toString();
```
Comment (J-N-K) [inline bundles/org.openhab.core.ui.icon/src/test/java/org/openhab/core/ui/icon/internal/IconServletTest.java:69]: Is this guaranteed to be the same?

**Review by J-N-K — COMMENTED** [review]
Otherwise LGTM

File: bundles/org.openhab.core.ui.icon/src/test/java/org/openhab/core/ui/icon/internal/IconServletTest.java (hunk 1 above)
Comment (wborn) [reply bundles/org.openhab.core.ui.icon/src/test/java/org/openhab/core/ui/icon/internal/IconServletTest.java:69]: Yes it saves creating a copy of the byte array:

## ByteArrayOutputStream

```java
public synchronized byte[] toByteArray() {
    return Arrays.copyOf(buf, count);
}

public synchronized String toString() {
    return new String(buf, 0, count);
}
```

## String

```java
public String(byte[] bytes) {
    this(bytes, 0, bytes.length);
}
```

**Review by J-N-K — APPROVED** [review]
Thanks!

**Pull request merged**